static/app/components/searchQueryBuilder/valueCombobox.tsx

```diff
@@ -0,0 +1,116 @@
+import {useMemo, useState} from 'react';
+import {Item, Section} from '@react-stately/collections';
+
+import {getItemsWithKeys} from 'sentry/components/compactSelect/utils';
+import {SearchQueryBuilderCombobox} from 'sentry/components/searchQueryBuilder/combobox';
+import {useSearchQueryBuilder} from 'sentry/components/searchQueryBuilder/context';
+import {
+  escapeTagValue,
+  formatFilterValue,
+} from 'sentry/components/searchQueryBuilder/utils';
+import type {Token, TokenResult} from 'sentry/components/searchSyntax/parser';
+import type {SearchGroup} from 'sentry/components/smartSearchBar/types';
+import {t} from 'sentry/locale';
+import type {Tag} from 'sentry/types';
+import {FieldValueType, getFieldDefinition} from 'sentry/utils/fields';
+import {type QueryKey, useQuery} from 'sentry/utils/queryClient';
+
+type SearchQueryValueBuilderProps = {
+  token: TokenResult<Token.FILTER>;
+};
+
+function isStringFilterValues(
+  tagValues: string[] | SearchGroup[]
+): tagValues is string[] {
+  return typeof tagValues[0] === 'string';
+}
+
+function getPredefinedValues({key}: {key?: Tag}): string[] {
+  if (!key) {
+    return [];
+  }
+
+  const fieldDef = getFieldDefinition(key.key);
+
+  if (!key.values) {
+    return [];
+  }
+
+  if (isStringFilterValues(key.values)) {
+    return key.values;
+  }
+
+  switch (fieldDef?.valueType) {
+    // TODO(malwilley): Better duration suggestions
+    case FieldValueType.DURATION:
+      return ['-1d', '-7d', '+14d'];
+    case FieldValueType.BOOLEAN:
+      return ['true', 'false'];
+    // TODO(malwilley): Better date suggestions
+    case FieldValueType.DATE:
+      return ['-1h', '-24h', '-7d', '-14d', '-30d'];
+    default:
+      return [];
+  }
+}
+
+export function SearchQueryBuilderValueCombobox({token}: SearchQueryValueBuilderProps) {
+  const [inputValue, setInputValue] = useState('');
+
+  const {getTagValues, keys, dispatch} = useSearchQueryBuilder();
+  const key = keys[token.key.text];
+  const shouldFetchValues = key && !key.predefined;
+
+  // TODO(malwilley): Display error states
+  const {data} = useQuery<string[]>({
+    queryKey: ['search-query-builder', token.key, inputValue] as QueryKey,
+    queryFn: () => getTagValues(key, inputValue),
+    keepPreviousData: true,
+    enabled: shouldFetchValues,
+  });
+
+  const items = useMemo(() => {
```

Comment:
If the field has `predefined: true` we return a list of values. Otherwise we call `getTagValues` that is passed in. This matches the current behavior in SmartSearchBar